spanner/api/Program.cs

```diff
@@ -928,34 +1091,55 @@ internal class CustomTransientErrorDetectionStrategy
             return ExitCode.Success;
         }
 
-        public static object QueryDataWithTransaction(string projectId,
-            string instanceId, string databaseId)
+        public static object QueryDataWithTransaction(string projectId, string instanceId, string databaseId, string platform)
         {
+
             var retryPolicy =
                 new RetryPolicy<CustomTransientErrorDetectionStrategy>
                     (RetryStrategy.DefaultExponential);
-            var response = retryPolicy.ExecuteAsync(async () =>
-               await QueryDataWithTransactionAsync(
-                    projectId, instanceId, databaseId));
+
+            var response = platform == NetCorePlatform
+                ? retryPolicy.ExecuteAsync(() =>
+                    QueryDataWithTransactionCoreAsync(projectId, instanceId, databaseId))
+                : retryPolicy.ExecuteAsync(() =>
+                    QueryDataWithTransactionAsync(projectId, instanceId, databaseId));
+
             s_logger.Info("Waiting for operation to complete...");
             response.Wait();
             s_logger.Info($"Operation status: {response.Status}");
             return ExitCode.Success;
         }
 
-        public static object ReadWriteWithTransaction(string projectId,
-            string instanceId, string databaseId)
+        public static object ReadWriteWithTransaction(string projectId, string instanceId, string databaseId, string platform)
         {
             try
             {
-                var retryPolicy =
-                    new RetryPolicy<CustomTransientErrorDetectionStrategy>
-                        (RetryStrategy.DefaultExponential);
-                var response = retryPolicy.ExecuteAsync(async () =>
-                await ReadWriteWithTransactionAsync(
-                    projectId, instanceId, databaseId));
                 s_logger.Info("Waiting for operation to complete...");
-                Task.WaitAll(response);
+                var response = platform == NetCorePlatform
+                    ? Task.Run(async () =>

```

Comment:
I like how you put the 4.6 vs core code in two separate functions, like you did on line 1103.  Can you do that here too?